sunpy/net/helioviewer.py

```diff
@@ -75,11 +105,23 @@ def get_closest_image(self, date, **kwargs):
         >>> print(metadata['date'])  # doctest: +REMOTE_DATA
         2012-01-01 00:00:07
         """
+        global datasource_info
+        
+        if('sourceId' in kwargs):
+            sourceid = kwargs['sourceId']
+        else:
+            key = (observatory, instrument, detector, measurement)
+            sourceid = datasource_info[key]
+
         params = {
             "action": "getClosestImage",
-            "date": self._format_date(date)
+            "date": self._format_date(date),
+            "sourceId": sourceid,
+            "observatory": observatory,
+            "instrument": instrument,
+            "detector": detector,
+            "measurement": measurement
         }
-        params.update(kwargs)
```

Comment:
Don't we still want this?

Reply:
Yes, because right now we are allowing the user to use sourceID in kwargs like before. We can remove this if we restrict the user to use sourceID directly. But as per the older version of the helioviewer API the user was allowed to pass in the source ID to get the image. So now it depends on us if we want the user to use source ID or not

Reply:
So we need to add it back?

Reply:
Um we can remove it given that we don't allow the user to pass in any extra parameters like source id

Reply:
I think if we have this in other functions, it should stay here.

Reply:
Ok I will see that

Reply:
This should be added back please.